scripts/pylib/twister/twisterlib.py

```diff
@@ -2152,8 +2151,9 @@ def set_case_status_by_name(self, name, status, reason=None):
             tc.reason = reason
         return tc
 
-    def add_testcase(self, name):
+    def add_testcase(self, name, freeform=False):
         tc = TestCase(name=name)
+        tc.freeform = freeform
```

Comment:
As I mentioned earlier, `freeform` argument could be passed with `__init__` function:
```python
tc = TestCase(name=name, testsuite=self, freeform=freeform)
```